espnet2/enh/loss/criterions/time_domain.py

```diff
@@ -20,10 +20,37 @@ def name(self) -> str:
     def only_for_test(self) -> bool:
         return self._only_for_test
 
-    def __init__(self, name, only_for_test=False):
+    @property
+    def is_noise_loss(self) -> bool:
+        return self._is_noise_loss
+
+    @property
+    def is_dereverb_loss(self) -> bool:
+        return self._is_dereverb_loss
+
+    def __init__(
+        self,
+        name,
+        only_for_test=False,
+        is_noise_loss=False,
+        is_dereverb_loss=False,
+    ):
         super().__init__()
-        self._name = name
+        # only used during validation
         self._only_for_test = only_for_test
+        # only used to calculate the noise-related loss
+        self._is_noise_loss = is_noise_loss
+        # only used to calculate the dereverberation-related loss
+        self._is_dereverb_loss = is_dereverb_loss
```

Comment:
I saw you added `only_for_test`, `is_noise_loss ` and `is_dereverb_loss ` for both time-domain loss and frequency-domain loss.  Adding these attributes to `AbsEnhLoss(torch.nn.Module, ABC)`  might be a more brief way. The current approach is good. It's just a suggestion.

Reply:
My initial thought is that some losses may not be designed for noise signals, like the DPCL loss. So I didn't add these properties to `AbsEnhLoss` directly.
If you think it is better to inherit from `AbsEnhLoss`, I would do that.